text_edit auto_completion

AlexHolly opened the pull request:
replaced similarity with begins_with

closes https://github.com/godotengine/godot/issues/16045
New version(you can see the old in the issue)
![peek 2018-07-29 15-47](https://user-images.githubusercontent.com/4741886/43371096-4478ba0c-938b-11e8-8746-594532e352e8.gif)
![peek 2018-07-30 00-01](https://user-images.githubusercontent.com/4741886/43371131-ca1805be-938b-11e8-8698-11b2370d0a42.gif)

This is something I wasn't looking for but it seems to work better now(still not perfect).
Old version
![peek 2018-07-30 00-27](https://user-images.githubusercontent.com/4741886/43371316-6820a948-938f-11e8-88ba-4ab2395b95ac.gif)
New version
![peek 2018-07-30 00-30](https://user-images.githubusercontent.com/4741886/43371340-db831312-938f-11e8-942b-5663c3e22836.gif)







related? https://github.com/godotengine/godot/issues/20565

vnen commented:
What happen if you type something that's not an exact match?

AlexHolly commented:
Example - "Not exact"
looking for `OS.get_executable_path`
![peek 2018-07-30 12-55](https://user-images.githubusercontent.com/4741886/43393556-e947a45a-93f7-11e8-9b87-c7e82569c2e7.gif)

the load keyword is missing as a keyword in the docs.
http://docs.godotengine.org/en/latest/getting_started/scripting/gdscript/gdscript_basics.html

It looks like `preload` is missing `'('` and they both(load/preload) don't activate the auto_completion like `connect` does now.

![peek 2018-07-30 13-22](https://user-images.githubusercontent.com/4741886/43394600-9279f106-93fb-11e8-945c-14d11bae2d06.gif)

akien-mga commented:
Could you squash commits together and edit the commit log to be more descriptive?

AlexHolly committed:
text_edit auto_completion

replaced "similarity filter" with "sequential checks" this give more control over the desired filtering.

akien-mga commented:
Still a weird commit log, but code looks fine, thanks :)
